testing/code/test_code.py

```diff
@@ -67,12 +68,8 @@ def func():
 
     f = func()
     f = _pytest._code.Frame(f)
-    prop = f.code.__class__.fullsource
-    try:
-        f.code.__class__.fullsource = None
-        assert f.statement == _pytest._code.Source("")
-    finally:
-        f.code.__class__.fullsource = prop
+    with mock.patch.object(f.code.__class__, "fullsource", None):
```

Comment:
this should use monkeypatch

Reply:
oops didn't see this until already merged -- `monkeypatch` has a much wider scope than the code I was replacing and so it is not equivalent.

(And since this is monkeypatching a pytest internal I don't really want that to leak out of this test)

Curious also why to prefer `monkeypatch` over `mock.patch` -- the latter is much more battle tested and has a well defined patch scope due to the with block